### ISSUE TITLE
Issue with UTF 8 

### ISSUE THREAD

**Issue by jas1** [ropensci/RefManageR#40]
Session INFO:

> 
> > sessionInfo()
> R version 3.4.1 (2017-06-30)
> Platform: x86_64-w64-mingw32/x64 (64-bit)
> Running under: Windows >= 8 x64 (build 9200)
> 
> Matrix products: default
> 
> locale:
> [1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252    LC_MONETARY=Spanish_Spain.1252 LC_NUMERIC=C                  
> [5] LC_TIME=Spanish_Spain.1252    
> 
> attached base packages:
> [1] stats     graphics  grDevices utils     datasets  methods   base     
> 
> other attached packages:
> [1] RefManageR_0.14.20 bindrcpp_0.2       rebus_0.1-3        stringi_1.1.5      stringr_1.2.0      dplyr_0.7.1       
> [7] rvest_0.3.2        xml2_1.1.1        
> 
> loaded via a namespace (and not attached):
>  [1] Rcpp_0.12.12          bindr_0.1             magrittr_1.5          R6_2.2.2              rlang_0.1.1          
>  [6] bibtex_0.4.2          plyr_1.8.4            httr_1.2.1            tools_3.4.1           rebus.numbers_0.0-1  
> [11] assertthat_0.2.0      tibble_1.3.3          rebus.datetimes_0.0-1 glue_1.1.1            rebus.base_0.0-3     
> [16] compiler_3.4.1        rebus.unicode_0.0-2   jsonlite_1.5          lubridate_1.6.0       pkgconfig_2.0.1     
> 

Context details: 

* working on spanish references ( accents / tildes / ñ, etc )
* have several bib files that are UTF-8 with bom ( checked with notepad++ )
* i aggregate the bib metadata to a dataframe then save the dataframe
* read the file with the following code ( example for only one )

```
## articulo_bib > is the path to the file 
  bibo <- ReadBib(articulo_bib,.Encoding = "UTF-8")
  btype <- bibo$bibtype
  baut <- paste( bibo$author, collapse = ';')
  bjour <- bibo$journal
  btitle <- bibo$title
```
as the author names , abstrct, title, even journal name,  have special characters, 
it reads weird the data: 

if i just execute  the " bibo " variable , it returns : 
```
[1] C. Aranda and L. Luna. “Enfoques actuales y problemáticas multidisciplinares de la bioarqueología contemporánea
en Argentina. Introducción y conclusiones del simposio / Current approaches and multidisciplinary issues of
contemporary bioarchaeology in Argentina. Introduction and conclusions of the symposium”. In: _Revista Argentina de
AntropologÃ­a BiolÃ³gica_ (2012), pp. 3-8. ISSN: 1853-6387. <URL:
https://revistas.unlp.edu.ar/raab/article/view/612>.
```

as you can see it have the special characters on some places, but in other places it returns weird characters.

when i print out the journal name, it wont work properly:

```
> bibo$journal
[1] "Revista Argentina de AntropologÃ­a BiolÃ³gica"
```

same for title:

```
> bibo$title
[1] "Enfoques actuales y problemÃ¡ticas multidisciplinares de la bioarqueologÃ­a contemporÃ¡nea en Argentina. IntroducciÃ³n y conclusiones del simposio / Current approaches and multidisciplinary issues of contemporary bioarchaeology in Argentina. Introduction and conclusions of the symposium"
```

for temporal fix i managed to convert the characters with the following lines:
( may be can save time to others )

```
baut <- iconv( baut, "UTF-8", "latin1")
btitle <- iconv( bibo$title, "UTF-8", "latin1")
bjour <- iconv( bibo$journal, "UTF-8", "latin1")
```

after this change, the i can save the data to an UTF-8 file , and it worked like charm.

```
## acumulado_bib_tex: its the dataframe that aggregates the several references im working on
## full_path_bib_tex: its the place where i want to write the aggregated data frame
write.table(acumulado_bib_tex, full_path_bib_tex,sep = '\t',row.names = FALSE,quote = TRUE,fileEncoding = 'UTF-8') 
```

Hope you can manage to fix the issue, so no temporal fix is needed :D

**Comment by mwmclean** [comment]
Did you try using `Sys,setlocale` to change from "Spanish_Spain.1252" to e.g. "es_utf8"?

**Comment by jas1** [comment]
Hi mwmclean; i'm working from several computers, 1 linux , the other 2 in windows, guess that brought some issues / confussion on the subject to me.

I tryed to use the sys.setlocale() but guess I got it wrong as it did not worked.
At the moment I dont have the sys.setlocale that I tryed. 

My reference was:  https://stat.ethz.ch/R-manual/R-devel/library/base/html/locales.html
Nevertheless,  the solution did not convinced me as it has several warnings.

Also got a doubt about: where could I get a list with the valid windows locales ? 
( _if you have a source I'll be really gratefull_ ) 

For now I removed the iconv stuff, as now im using pure utf-8 at linux. 

After realized the win/linux issue, I just prefered to work on the linux machine for the dataset generation; as i was having other issues too. As I generate the bibtex files form html files, save them as UTF-8 .bib files; then read all of them with the RefmanageR; to produce a data.frame with the values i need from the files. ( _having issues on reading the generated dataframe because of escaping quotes, and other characters._ ) 

For now I keep working on the linux pc ; Later on might get back to the issue, to see if i get with a better solution / time to spend on this task. ( _dont know if i can comment after closing_ )

Will be awsome to have a more straight forward way to read / save defining the encoding dont relying on the sys.setlocale.  ( _dont know if it can be done_ )

Thanks for your time =)
